src/core/ext/lb_policy/grpclb/grpclb.c

```diff
@@ -296,20 +300,53 @@ typedef struct glb_lb_policy {
    * response has arrived. */
   grpc_grpclb_serverlist *serverlist;
 
-  /** addresses from \a serverlist */
-  grpc_lb_addresses *addresses;
-
   /** list of picks that are waiting on RR's policy connectivity */
   pending_pick *pending_picks;
 
   /** list of pings that are waiting on RR's policy connectivity */
   pending_ping *pending_pings;
 
-  /** client data associated with the LB server communication */
-  struct lb_client_data *lb_client;
+  bool shutting_down;
+
+  /************************************************************/
+  /*  client data associated with the LB server communication */
+  /************************************************************/
+  /* called once initial metadata's been sent */
+  grpc_closure md_sent;
+
+  /* called once the LoadBalanceRequest has been sent to the LB server. See
+   * src/proto/grpc/.../load_balancer.proto */
+  grpc_closure req_sent;
+
+  /* A response from the LB server has been received (or error). Process it */
+  grpc_closure res_rcvd;
```

Comment:
A few naming suggestions:

- Please prefix all of the fields that were formerly in the `lb_client_data` struct with `lb_`, so that it's clear which ones are related to communication with the LB.
- Please avoid abbreviations (e.g., prefer `lb_response_received()` to `res_rcvd()`).
- I suggest using the same name for each closure and the corresponding callback function.